File: OpenCV/pythonTest/drawPoly.py
import cv
import numpy as np

def main():
    red = (0, 0, 255)
    green = (0, 255, 0)
    img = np.full((400, 540, 3), 255, dtype = np.uint8)

    a = 0

    while True:

        cv2.line(img, (10, 10), (100 + a, 100 + a), red, 3)
        cv2.imshow("img", img)
        
        if cv2.waitKey(0) == 27:
            break

        a += 1

if __name__ == "__main__":
    main()

File: OpenCV/cppTest/drawPoly.cpp
#include "opencv2/opencv.hpp"
#include <iostream>

using namespace cv;
using namespace std;

int main()
{
    Scalar green = Scalar(0, 255, 0);
    Scalar red = Scalar(0, 0, 255);

    Mat img(400, 640, CV_8UC3, Scalar(255, 255, 255));
    int a = 0;

    while(true) {
        img = Scalar(255, 255, 255); // 배경 흰색으로 초기화

        rectangle(img, Rect(50 + a, 50 + a, 100, 100), green, 5);
        circle(img, Point(500 - a, 100), 50, red, -1);
        ellipse(img, Point(100 + a, 300), Size(20, 10), 30, 0, 180, green, 5);

        a++;

        if (a > 400)
            a = 0;

        imshow("img", img);

        if (waitKey(10) == 27)
            break;
    }

    return 0;
}